### DISCUSSION TITLE
CPU load

### DISCUSSION THREAD

**Discussion by ssypchenko** [weltenwort/frigate-synology-dsm7#67] [General]
Guys, do you have any ideas why Synology Surveillance Station with Deepstack and SynoAI uses 20% CPU, but Frigate with CPU detection uses 80% CPU? I understand that  Coral is preferable to use with Frigate. But SSS do the same job with CPU detection and 4 times less utilisation...

**Comment by baylanger** [comment]
I wasn't aware of SynoAI / Deepstack.  Interesting project.

To answer you great question... makes me wondering as well why.  Right now, I've got no idea .. just writing what comes.

DS920+ (20GB RAM) - Deepstack set to "low" - ~2 seconds for image recognition
DS918+ (12GB RAM) - Deepstack set to "medium" - ~4 seconds for image recognition
DS713+ (4GB RAM) - Deepstack set to "low" - ~9 seconds for image recognition

^^ Ref : https://github.com/Christofo/sssAI

Is SSS multi threaded?  What about Frigate?

Is SSS called only when Surveillance Station triggers an alert?  Frigate is analyzing the image non-stop and SSS might not be doing this.  Would need confirmation.

**Comment by weltenwort** [comment]
The CPU usage on my system mostly stems from ffmpeg resizing the video stream from one of my cameras, because the device can't be configured to server an appropriate stream. Maybe you can check which process uses the CPU in your case. (see https://github.com/blakeblackshear/frigate/discussions/2426#discussioncomment-2930367)

Also: thanks for bringing SynoAI to my attention, looks interesting!

**Comment by ssypchenko** [comment]
Yes, SSS calls Deepstack only when it detected a motion to recognise the object. So it takes much less CPU time for image analysis. 
I checked the processes and see that 30-50% of utilization is from the "cpu.detector" process. 
One more nice feature of SSS is that it records and stores the low-resolution video for stable pictures and starts recording high resolution if it detects motion. It saves a lot of space on NAS. 
So I considered Frigate just because not have to pay for licenses. But from other points of view, SSS looks better as I personally see it. 

This topic was not about the issue. I just wanted to share my thoughts with professionals about different options with CCTV on Synology. So you can close the issue if necessary.

**Reply by baylanger** [reply]
> One more nice feature of SSS is that it records and stores the low-resolution video for stable pictures and starts recording high resolution if it detects motion. It saves a lot of space on NAS.

Best idea I heard lately :)

Is there a video format that allows to switch resolution on-the-fly w/o the need to use another file?  I don't think so... SSS might be saving 2 different files (low/high res) and when streaming, depending on what the resolution requested by the client, it manages to convert when needed.  I'm not sure how else they could manage to do this.

**Reply by baylanger** [reply]
@ssypchenko I'm curious... why [SSS](https://github.com/Christofo/sssAI) over [SynoAI](https://github.com/djdd87/SynoAI) ?

**Reply by ssypchenko** [reply]
> @ssypchenko I'm curious... why [SSS](https://github.com/Christofo/sssAI) over [SynoAI](https://github.com/djdd87/SynoAI) ?

SSS (Synology Surveillance Station) receives video from cameras and detects motions. SynoAI works as a bridge between SSS and Deepstack and sends notifications to telegram. So I switched off all notifications in SSS to avoid sun, shadows and wind notifications, and I get only pictures with objects selected to telegram from Deepstack.

**Comment by weltenwort** [comment]
Interesting - as far as I understand frigate it also waits for a motion trigger before it starts feeding the video to the inference engine. But I agree wiring up the native motion detection as a trigger would be a valuable optimization. Maybe something like that can be added to frigate too. :thinking:

I'll convert this into a discussion in case others stop by and have something to contribute.
